applications/ContactStructuralMechanicsApplication/custom_strategies/custom_convergencecriterias/base_mortar_criteria.h

```diff
@@ -88,7 +88,7 @@ class BaseMortarConvergenceCriteria
     BaseMortarConvergenceCriteria(const bool IODebug = false)
         : ConvergenceCriteria< TSparseSpace, TDenseSpace >(),
           mIODebug(IODebug),
-          mGidIO("POST_LINEAR_ITER", GiD_PostBinary, SingleFile, WriteUndeformed,  WriteElementsOnly)
+          mGidIO("POST_LINEAR_ITER", GiD_PostBinary, SingleFile, WriteUndeformed,  WriteConditions)
```

Comment:
i find very strange that you pass the gidio object to the convergence criteria...

Reply:
It is just for debugging during non-linear iterations

Reply:
at least default it to null... and handle the case in which this is not provided